include/kamping/data_buffer.hpp

```diff
@@ -586,21 +598,21 @@ class DataBuffer : private CopyMoveEnabler<enable_copy_construction_v<ownership>
     ///
     /// @return Moves the underlying container out of the DataBuffer.
     template <bool enabled = is_owning, std::enable_if_t<enabled, bool> = true>
-    MemberTypeWithConst extract() {
+    StorageType extract() {
         static_assert(
             ownership == BufferOwnership::owning,
             "Moving out of a reference should not be done because it would leave "
             "a users container in an unspecified state."
         );
         kassert_not_extracted("Cannot extract a buffer that has already been extracted.");
-        auto extracted = std::move(underlying());
+        auto extracted = std::move(_data);
```

Comment:
ensure that we also call the `vector<bool>` static assert from `underlying()` here